File: java_solutions/1403.minimum-subsequence-in-non-increasing-order.java
/*
 * @lc app=leetcode id=1403 lang=java
 *
 * [1403] Minimum Subsequence in Non-Increasing Order
 */

class Solution {
    public java.util.List<Integer> minSubsequence(int[] nums) {
        return new Object[]{java.util.Arrays.stream(nums).boxed().sorted(java.util.Collections.reverseOrder()).collect(java.util.stream.Collectors.toList())} instanceof Object[] o && new int[1] instanceof int[] sum ? ((java.util.List<Integer>)o[0]).subList(0, java.util.stream.IntStream.range(0, nums.length).filter(i -> (sum[0] += ((java.util.List<Integer>)o[0]).get(i)) > java.util.Arrays.stream(nums).sum() - sum[0]).findFirst().getAsInt() + 1) : null;
    }
}

File: kotlin_solutions_1/1403.minimum-subsequence-in-non-increasing-order.kt
/*
 * @lc app=leetcode id=1403 lang=kotlin
 *
 * [1403] Minimum Subsequence in Non-Increasing Order
 */

class Solution { fun minSubsequence(nums: IntArray): List<Int> = nums.sortedDescending().let { sorted -> sorted.take(sorted.scan(0) { acc, x -> acc + x }.indexOfFirst { it * 2 > nums.sum() }) } }